scipy/optimize/_minimize.py

```diff
@@ -568,7 +568,8 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
     if meth in ('nelder-mead', 'l-bfgs-b', 'tnc', 'powell') and np.any(constraints):
         warn('Method %s cannot handle constraints.' % method,
              RuntimeWarning)
-    if meth == 'cobyla' and bounds is not None:
+    if meth not in ('nelder-mead', 'powell', 'l-bfgs-b', 'tnc', 'slsqp',
+                    'trust-constr') and bounds is not None:
```

Comment:
```suggestion
                    'trust-constr', '_custom') and bounds is not None:
```

Reply:
mmmm this PR should not have to do anything with that. Could you merge `main` into your PR?